src/RageFileBasic.cpp

```diff
@@ -84,145 +84,145 @@ int RageFileObj::Seek( int iOffset )
 	 * flush before seeking to do proper error checking. */
 	ResetReadBuf();
 
-	int iPos = SeekInternal( iOffset );
-	if( iPos != -1 )
-	        m_iFilePos = iPos;
+	int iPos = SeekInternal(iOffset);
+	if (iPos != -1)
+		m_iFilePos = iPos;
 	return iPos;
 }
 
-int RageFileObj::Seek( int offset, int whence )
+int RageFileObj::Seek(int offset, int whence)
 {
-	switch( whence )
+	int iOffset = static_cast<int>(offset);
+	switch (whence)
 	{
 	case SEEK_CUR:
-		return Seek( Tell() + offset );
+		return Seek(Tell() + iOffset);
 	case SEEK_END:
-		offset += GetFileSize();
+		iOffset += GetFileSize();
 	}
-	return Seek( (int) offset );
+	return Seek(iOffset);
 }
 
-int RageFileObj::Read( void *pBuffer, std::size_t iBytes )
+int RageFileObj::Read(void* pBuffer, std::size_t iBytes)
 {
-	int iRet = 0;
+	char* pDest = static_cast<char*>(pBuffer);
```

Comment:
Same thing here. Doesn't seem like we need this local variable, esp since it points to the same memory address.

Reply:
The point of this is to prevent pBuffer from being cast to a char *. The idea here was that the contents of pBuffer would not be modified thus allowing weird characters to work with the game.